### PR TITLE
MacOSX installation refresh

### LINKED ISSUE #55: Revise dependency configuration instructions
This issue stems from the VTK compilation instructions where the MPI option `-DVTK_Group_MPI:BOOL=ON` is incorrect per the VTK build documentation. MPI options include `-DVTK_USE_MPI=ON` and `-DVTK_GROUP_ENABLE_MPI=YES`. The instructions should be cleaned up so that VTK is installed appropriately. I will post the various issues I had when turning on these MPI options below.

### PR DESCRIPTION
**Pull request by zhardy-lanl**
MacOSX installation refresh, including `configure_dependencies.py` modifications. 

Renamed `doc.*md` files to match snake_case naming convention.

Closes #55

### CODE REVIEW

File: resources/configure_dependencies.py [hunk 1]
```diff
@@ -138,7 +152,7 @@ def ExecSub(command: str,
 make_command = "make"
 s, e, outstr = ExecSub(make_command + " --version")
 if outstr.find("GNU Make") >= 0:
-    make_command = "make OMAKE_PRINTDIR=gmake"
+    make_command = "make OMAKE_PRINTDIR=make"
```
Comment (ragusa) [inline resources/configure_dependencies.py:155]: can you remind me what the issue was between make and gmake here? Thanks

File: resources/configure_dependencies.py [hunk 2]
```diff
@@ -538,34 +558,38 @@ def InstallVTK(pkg: str, ver: str, gold_file: str):
         os.chdir("chi_build")
```
Comment (ragusa) [inline resources/configure_dependencies.py:559]: still chi_build here??

File: resources/configure_dependencies.py (hunk 1 above)
Comment (zhardy-lanl) [reply resources/configure_dependencies.py:155]: There is no issue. This fix came from what PETSc support sent me long ago when I was unable to build PETSc. Theirs used `gmake` for both, I have always used `make` for both and it works fine. This was just done for consistency.

File: resources/configure_dependencies.py (hunk 2 above)
Comment (zhardy-lanl) [reply resources/configure_dependencies.py]: Should be fixed now. Missed a couple.

**Review by ragusa — APPROVED** [review]

**Review by andrsd — CHANGES_REQUESTED** [review]
Overall good changes with a few minor things.

Also, please, add a patch on top of this branch that will rename:
```
Install_macos.md -> install_macos.md
Install_macos_easy.md -> install_macos_easy.md
```
So, the file names conform to the naming convention.

File: doc/Install_macos.md [hunk 3]
```diff
@@ -1,81 +1,88 @@
...
-    Target: x86_64-apple-darwin18.7.0
-    Thread model: posix
-    InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
+export CC="${MPI_DIR}/mpicc"
+export CXX="${MPI_DIR}/mpicxx"
+export FC="${MPI_DIR}/mpifort"
+export F77="${MPI_DIR}/mpif77"
 ```
+to the `~/.bash_profile` or `~/.bashrc` script. Make sure to execute the script
+to load the changes.````
```
[the remark this answers is not in the document]
Comment (zhardy-lanl) [reply doc/Install_macos.md:61]: Good catch.

File: doc/Install_macos.md [hunk 4]
```diff
@@ -88,138 +95,139 @@ CXXOPTFLAGS='-O3 -march=native -mtune=native'  \
...
+mkdir build
+cd build
+cmake ..
 ```
-The configure script will generate the CMake build scripts.
-
-Once configure.sh has completed, build **Chi-Tech** with:
-```console
-    $ make
+The script will generate the CMake build scripts.
```
[the remark this answers is not in the document]
Comment (zhardy-lanl) [reply doc/Install_macos.md:197]: Recycled old language. Changing to your suggestion.

File: doc/Install_macos.md [hunk 5]
```diff
@@ -88,138 +95,139 @@ CXXOPTFLAGS='-O3 -march=native -mtune=native'  \
...
 
-### Step 8 - Chi-Tech documentation
-
-You can either access the documentation online [here](https://chi-tech.github.io), or generate it locally.
-
-To generate the documentation from your local working copy, first make sure
-Doxygen and LaTeX are installed:
-
-```bash
+The documentation can be found [online](https://chi-tech.github.io), or
```
[the remark this answers is not in the document]
Comment (zhardy-lanl) [reply doc/Install_macos.md:216]: I suppose so. This was just reformatting of what was already there.

**Review by andrsd — APPROVED** [review]

**Comment by zhardy-lanl** [comment]
Please do not merge until a small change comes through. There is a small issue with the Lua compilation that causes some of the `framework/parameters` tests to fail. Trying to find a solution now.

**Comment by zhardy-lanl** [comment]
The issue mentioned above is that `test/framwork/parameters/params_01c/d.lua` fail due to the error message not displaying properly. It appears as though the intended error gets caught, but the error handling does not occur as expected. The exact cause of this is unknown, but compiling lua with `gcc` instead of `mpich` removes this issue and the tests pass once again.

**Pull request merged**